### PR TITLE
Improve, but not perfect, Posix pthread code quality.

### PR DESCRIPTION
**Pull request by LeeTibbert**
We ameliorate Posix pathread code quality by removing `@name`
annotations in pthread.scala which are both unnecessary and have
not corresponding code in pthread.c.

A name is changed in both pthread.scala & pthread.c to make it parallel
similar names.

Lastly, `scalanative_pthread_once_init()` in pthread.c is simplified
to make it parallel other constant declarations, to show that it is
exactly similar to those, and to make it easier to trace & understand.

### CODE REVIEW

**Comment by ekrich** [comment]
Since this was created before we had standards there are quite a few irregularities - https://scala-native.readthedocs.io/en/latest/contrib/contributing.html#c-posix-libraries

Also, note that some of the Scala functions don't have a corresponding function in the POSIX spec. These should probably be deprecated so we can remove them in the next release. https://pubs.opengroup.org/onlinepubs/9699919799/basedefs/pthread.h.html

**Comment by ekrich** [comment]
Given the comment [here](https://github.com/scala-native/scala-native/pull/2298#issuecomment-842692464) and that there wasn't even a `pthread_create` that worked, there is little chance that anyone is using this so unless someone disagrees, this PR doesn't need to deprecate or do anything special unless something is being used internal to Scala Native. 

@errikos, did you plan to submit a PR? If not perhaps you can give feedback here as you find things out.

File: posixlib/src/main/scala/scala/scalanative/posix/pthread.scala [hunk 1]
```diff
@@ -17,9 +18,9 @@ import scala.scalanative.posix.sys.types._
 // SUSv2 version is used for compatibility
 // see http://pubs.opengroup.org/onlinepubs/007908799/xsh/threads.html
 
+@link("pthread")
```
Comment (WojciechMazur) [inline posixlib/src/main/scala/scala/scalanative/posix/pthread.scala:21]: I think we should leave this explicit `link`. Yes, we do link pthread anyway, since it's needed in GC, however, it might not be actually needed for every GC, eg. none. If we would link pthread conditionally for GC uses and forget to restore this annotation the build would be broken (especially since it's not widely used)  What do you think?

Comment (LeeTibbert) [reply posixlib/src/main/scala/scala/scalanative/posix/pthread.scala:21]: @WojciechMazur  Thank you for reading through this PR.

You do a very good job of describing exactly the eason why this PR would
 like to _add_ the explicit `@link`. That line was not there before (understanding
that `@link` may not have been available when that code was written.)

**Comment by LeeTibbert** [comment]
@ekrich , @errikos 

Thank you both for your efforts on this PR.

##### TL;DR

To make the life of the current merge supervisor/gatekeeper easier, I believe that we
owe @WojciechMazur a clear consensus YES or NO on this PR, as it is. 

Lack consensus indicating a NO.   

Your thoughts?

##### Discussion:

To summarize:

1)  This PR, as-is, removes several "`@name` to no where", a potential build
      failure (by adding `@link1), and two "WT_???".  
  
     It makes the scene more presentable that it was before and does nothing to 
     either preclude or make future improvements more difficult.
     Conversely, it probably eases future work.

2) I believe that the removal of `pthread_` in `@name` annotations & C code
    should be done in a separate PR.  This would ease people understanding
    the changes of this PR.

    Ultimately, the separate PR discussion becomes a conversation between
    'lumpers' and "splitters'.   

   I am 85/15 for supporting the change.

    I can create an Issue so that the suggestion does not go missing.  I do not
    see myself having time in the foreseeable future to enter such a PR.

3) I think that other issues, such as moving useful (?) but non-Posix methods
    to separate files or changing declaration order  should be postponed
    for 6 months to a year, especially if @errikos is likely to be working
    in that area.

    Separate Issues could be created to capture the
    discussion and set a timer, in case, the proposed work takes
    a different direction and does not address these defects.

**Comment by LeeTibbert** [comment]
@ekrich 
re: 
```
Given the comment here and that there wasn't even a pthread_create that worked, there is little chance that anyone is using this so unless someone disagrees, this PR doesn't need to deprecate or do anything special unless something is being used internal to Scala Native.
```

I concur with your reasoning. Just did not have time to write it down before you did.
I had wondered/thought about those when I saw them, but whistled past them
in the drive to a small, svelte, & focused PR.

I think that any such work should be separate from this PR.

I naturally wondered "Why are these here??? (Well, more Anglo-Saxon that that)".
I had the flash "Probably to aid allocations of otherwise 'opaque' structures
and avoid declaring bunches of scalanative_foo structures which break opaquenes)
Interrupted.  Gotta run, sorry.

File: posixlib/src/main/scala/scala/scalanative/posix/pthread.scala (hunk 1 above)
Comment (WojciechMazur) [reply posixlib/src/main/scala/scala/scalanative/posix/pthread.scala:21]: Right, I have no idea why I've confused that with deletion! :D Sorry for bothering you with that.

**Comment by ekrich** [comment]
I think it might be better to kick the can down the road or if @errikos uses `pthread` then perhaps he might want to submit a PR.

If you want to merge this, I have a couple of small change requests. I will go ahead and do that review anyway just in case.

File: posixlib/src/main/scala/scala/scalanative/posix/pthread.scala [hunk 2]
```diff
@@ -294,7 +290,7 @@ object pthread {
   @name("scalanative_size_of_pthread_t")
   def pthread_t_size: CSize = extern
 
-  @name("scalanative_pthread_attr_t")
+  @name("scalanative_size_of_pthread_attr_t")
   def pthread_attr_t_size: CSize = extern
```
Comment (ekrich) [inline posixlib/src/main/scala/scala/scalanative/posix/pthread.scala:294]: This should be `@name("scalanative_pthread_attr_t_size")` to match our conventions and then the C file too.

If you could do that for all these helpers to make sure they match then I would be fine with these enhancements.

File: posixlib/src/main/resources/scala-native/pthread.c [hunk 3]
```diff
@@ -20,51 +20,46 @@ size_t scalanative_size_of_pthread_mutexattr_t() {
...
-int scalanative_pthread_mutex_recursive() { return PTHREAD_MUTEX_RECURSIVE; }
+int scalanative_mutex_recursive() { return PTHREAD_MUTEX_RECURSIVE; }
 
-pthread_once_t scalanative_pthread_once_init() {
-    pthread_once_t once_block = PTHREAD_ONCE_INIT;
-    return once_block;
-}
+pthread_once_t scalanative_once_init() { return PTHREAD_ONCE_INIT; }
 
-int scalanative_pthread_prio_inherit() { return PTHREAD_PRIO_INHERIT; }
```
Comment (ekrich) [inline posixlib/src/main/resources/scala-native/pthread.c:56]: Most of these like this one were correct. The C namespace is `PTHREAD_PRIO_INHERIT` so the Scala Native namespace is `scalanative_pthread_prio_inherit`. We are only prepending to the symbol. The exception is that we lowercase uppercase constants so the function in C is `c_like_function`. The reduces cognitive load for the reader as there is very little translation. Also, maintainers don't have to think things up.

Comment (LeeTibbert) [reply posixlib/src/main/resources/scala-native/pthread.c:56]: I made the corrections as you described.   

I believe the set of methods which _did_not_ have the `pthread_` were methods which
do not conform to the Posix spec.  The difference in name was probably to indicate that.

Anyway, I converted the names but left a comment block above the section, so that
adding the required `pthread_` does not lead to loss of the information that these are
not Posix.   

And of course, changing the names in the C file triggered the need to move the functions
so that the alphabetical order currently in the file is maintained.  Yes, at some point things
might get changed to declaration order. I think it _necessary_ that each iteration in the
evolution to perfection maintain _some_ internal consistency. Mainly because people
read in the now and may not be privy to future plans and those plans may never bear
fruit.

**Comment by LeeTibbert** [comment]
Passing the baton to @ekrich.

File: posixlib/src/main/resources/scala-native/pthread.c (hunk 3 above)
Comment (ekrich) [reply posixlib/src/main/resources/scala-native/pthread.c:56]: I think spec order is good - easier to compare.
